### DISCUSSION TITLE
Can´t Install Moonraker nor Nginx (Ender 3 KE) - SSL: CERTIFICATE_VERIFY_FAILED]

### DISCUSSION THREAD

**Discussion by macielc** [Guilouz/Creality-Helper-Script-Wiki#669] [Issues]
Greetings,

I´ve just tried to use your Creality Helper on my Ender 3 KE. 

As suggested, at first I uninstalled previous versions of mooranker, nginx and fluid using the menu "remove" function.

Once I tried to reinstall it, using your script, I got the following error with a crash:

rename moonraker/{ => components}/websockets.py (79%)
 create mode 100644 moonraker/utils/cansocket.py
 create mode 100644 moonraker/utils/exceptions.py
 create mode 100644 moonraker/utils/filelock.py
 create mode 100644 moonraker/utils/ioctl_macros.py
 create mode 100644 moonraker/utils/json_wrapper.py
 create mode 100644 moonraker/utils/pip_utils.py
 create mode 100644 moonraker/utils/sysfs_devs.py
 create mode 100644 moonraker/utils/versions.py
 delete mode 100644 pdm.lock
 create mode 100644 pdm_build.py
 create mode 100644 scripts/moonraker-dev-reqs.txt
 create mode 100644 scripts/moonraker-speedups.txt
 delete mode 100644 scripts/pdm_build_dist.py
 create mode 100644 scripts/python_wheels/zeroconf-0.131.0-py3-none-any.whl
 create mode 100755 scripts/sync_dependencies.py
Info: Installing Supervisor Lite...
Info: Installing Host Controls Support...
Info: Installing necessary packages...
WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: certificate is not yet valid (_ssl.c:1108)'))': /simple/pyserial-asyncio/
WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: certificate is not yet valid (_ssl.c:1108)'))': /simple/pyserial-asyncio/
WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: certificate is not yet valid (_ssl.c:1108)'))': /simple/pyserial-asyncio/
WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: certificate is not yet valid (_ssl.c:1108)'))': /simple/pyserial-asyncio/
WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: certificate is not yet valid (_ssl.c:1108)'))': /simple/pyserial-asyncio/
Could not fetch URL https://pypi.org/simple/pyserial-asyncio/: There was a problem confirming the ssl certificate: HTTPSConnectionPool(host='pypi.org', port=443): Max retries exceeded with url: /simple/pyserial-asyncio/ (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: certificate is not yet valid (_ssl.c:1108)'))) - skipping
ERROR: Could not find a version that satisfies the requirement pyserial-asyncio==0.6 (from versions: none)
ERROR: No matching distribution found for pyserial-asyncio==0.6
Could not fetch URL https://pypi.org/simple/pip/: There was a problem confirming the ssl certificate: HTTPSConnectionPool(host='pypi.org', port=443): Max retries exceeded with url: /simple/pip/ (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: certificate is not yet valid (_ssl.c:1108)'))) - skipping

Can you please help as my printer is now stucked and isolated?

Thank you,

Maciel

**Comment by herkne** [comment]
See the manual: **Also, make sure your system date and time are correct.** - https requires a correct date. Fix the date and retry.
